keras_rcnn/heads/resnet.py

```diff
@@ -0,0 +1,96 @@
+import keras
+
+
+def ResHead(classes, mask=False):
+    """Resnet heads as in Mask R-CNN."""
+    def f(x):
+        if keras.backend.image_data_format() == "channels_last":
+            channel_axis = 3
+        else:
+            channel_axis = 1
+
+        y = keras.layers.TimeDistributed(
```

Comment:
In my opinion, this is the trickiest part of the implementation.

I foresee two problems with using TimeDistributed.

- I suspect that in nearly every situation TimeDistributed is used to step across a temporal dimension. Unfortunately, we’re the rare exception. It’ll be a challenge to explain this in our documentation (especially if we expect users to re-implement the CNN that use for feature extraction to use TimeDistributed). I’d really like Keras to modify the name to better emphasize its utility.

- TimeDistributed requires users to implement their feature extractor layer by layer instead of calling on a model instance directly. I’d really prefer something like:

```Python
x = keras.layers.Input((224, 224, 3))

y = keras_resnet.ResNet50(x)

y = keras.layers.TimeDistributed(y)
```

If it worked like this, we could hide the `y = keras.layers.TimeDistributed(y)` statement inside the RCNN constructor. We could probably implement our own wrapper that iterates across a model’s layers and appends a TimeDistributed layer.